doc/guide/cert-authentication.xml

```diff
@@ -33,9 +33,11 @@
     pair and associate it to the user <code>alice</code>:</para>
 
 <programlisting>
-# create self-signed certificate and key:
+# create self-signed certificate and key; browsers insist on specifying key usage
 openssl req -x509 -newkey rsa:2048 -days 365 -nodes -keyout alice.key \
-    -out alice.pem -subj "/CN=alice"
+    -out alice.pem -subj "/CN=alice" -extensions v3_req \
+    -config &lt;(printf '[req]\ndistinguished_name=dn\nextensions=v3_req\n\
```

Comment:
I've never seen that syntax before.  Thanks for that.

That being said, I wonder if we could do something a little bit less magic here.  This is documentation and we probably shouldn't be aiming to teach our readers new shell tricks :-) 

Maybe just write to the config file separately as you do in the test?

Reply:
Ack, updated. I refreshed https://piware.de/tmp/cockpit-certauth/cert-authentication.html as well.